### PR TITLE
Speed up on go1.15 v2

### PR DESCRIPTION
**Pull request by c1rno**
The same as https://github.com/lamoda/gonkey/pull/58, but with different methods to shutdown, which accepts context from outside

### CODE REVIEW

File: mocks/service_mock.go [hunk 1]
```diff
@@ -38,10 +38,16 @@ func (m *ServiceMock) StartServer() error {
 	return nil
 }
 
+// Not used: backward compatibility with public interface
 func (m *ServiceMock) ShutdownServer() {
```
Comment (luza) [inline mocks/service_mock.go:42]: This is not a library-wide public method, so feel free to change its signature. Simply expand it with Context argument.

File: mocks/mocks.go [hunk 2]
```diff
@@ -41,10 +48,25 @@ func (m *Mocks) Start() error {
 	return nil
 }
 
+// Maybe slow after go1.15 for gracefull shutdown reason
+// https://github.com/golang/go/blob/release-branch.go1.15/src/net/http/server.go#L2679
+// See ShutdownContext for workaround
 func (m *Mocks) Shutdown() {
+	_ = m.ShutdownContext(context.TODO())
```
Comment (luza) [inline mocks/mocks.go:55]: I guess here we should pre-cancel the context to defeat the slowing down issue.

File: mocks/service_mock.go [hunk 3]
```diff
@@ -38,10 +38,16 @@ func (m *ServiceMock) StartServer() error {
...
-	m.server.Shutdown(context.TODO())
+	_ = m.ShutdownServerContext(context.TODO())
+}
+
+func (m *ServiceMock) ShutdownServerContext(ctx context.Context) (err error) {
```
Comment (luza) [inline mocks/service_mock.go:46]: Let's use unnamed return value, just like in all other methods around. Just to keep the coding style unified.

**Review by luza — CHANGES_REQUESTED** [review]
Thanks for your PR. This one looks good to me (I left a few comments though).

File: mocks/mocks.go (hunk 2 above)
Comment (c1rno) [reply mocks/mocks.go:55]: I planned to switch my tests to context-related method, and do it in my code, but ok, lets leave "old" behaviour here.

**Review by luza — APPROVED** [review]

**Review by What-If-I — APPROVED** [review]

**Pull request merged**